### PR TITLE
Updating rtofs to include modulefile and remove touch from prep

### PR DESCRIPTION
**Pull request by PerryShafran-NOAA**
## Pull Request Testing ##

- [ ] Describe testing already performed for this Pull Request:</br>

This PR:
- removes the module loads from the shell scripts and into a modulefile
- removes the touch command from the prep scripts
- does the tar of the plots in the directory the plots are in to ensure that the entire path of the file is not in the tarball
- updates the items for machine.conf

- [ ] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

Run the prep script at 07Z.  Ensure that you have at least 16 days worth of prep data (you may link to /lfs/h2/emc/vpppg/noscrub/perry.shafran/evs/v1.0/prep/rtofs (but check with me to be sure that all the data is there now that the machine is switching).  
Run all the stats jobs at 22Z.  
Run the plots jobs at 03Z the next morning.  

- [ ] Has the code been checked to ensure that no errors occur during the execution? **[Yes or No]**

Yes

- [ ] Do these updates/additions include sufficient testing updates? **[Yes or No]**

- [ ] Please complete this pull request review by **[Fill in date]**.</br>

ASAP.

## Pull Request Checklist ##

- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR description above.
- [ ] Ensure the PR title matches the feature branch name.
- [ ] Check the following:
- [ ]  Instructions provided on how to run
- [ ]  Developer's name is replaced by ${user} where necessary throughout the code
- [ ]  Check that the ecf file has all the proper definitions of variables
- [ ]  Check that the jobs file has all the proper settings of COMIN and COMOUT and other input variables
- [ ]  Check to see that the output directory structure is followed
- [ ]  Be sure that you are not using MET utilities outside the METplus wrapper structure

- [ ] After submitting the PR, select **Development** issue with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue.

### CODE REVIEW

**Comment by PerryShafran-NOAA** [comment]
Note - this PR does not yet include having the forecast hours of the plot files be three digits (i.e. f012) as requested by Alicia.  I will  need to get that done.  The other relevant parts of the rtofs changes are here, though, and can be tested, or you can wait until I get the 3-hr forecast hours added to this PR.

Perry

**Comment by AliciaBentley-NOAA** [comment]
Thank you, Perry! The 3 digit forecast hours can come in a subsequent PR.
Mallory tells me that GFS wave also used the wrong number of digits in the
images names and that she had to correct it when she took it over. Like you
said, I suspect that this is something that we will have to check for
across EVS.

___________________________________
*Alicia M. Bentley, Ph.D.*
Global Verification Project Lead
Model Evaluation Group (MEG) <https://www.emc.ncep.noaa.gov/users/meg/home/>
NOAA/NWS/NCEP/EMC (VPPPG Branch)
Work Phone: ‪(301) 683-3783‬

**Comment by ShelleyMelchior-NOAA** [comment]
@PerryShafran-NOAA -- can you please add `jevs_rtofs_prep.sh` to this PR to correct for the module loading happening from the sourced `modulefiles/rtofs/rtofs_prep.sh` file? I do not see that script included in this PR.

**Comment by PerryShafran-NOAA** [comment]
OK, this change has been made.  I'm not entirely sure why this file wasn't updated in the first place, but now it has.  

Sorry for the delay.

Perry

**Comment by AliciaBentley-NOAA** [comment]
Hi Perry,

I believe the path should say "modulefiles" and not "modules"
<https://github.com/NOAA-EMC/EVS/commit/107f7f5f21882a7b9920516aaa624f65b26ee8f4#diff-0c10e933e40c0e595027386eab3a97f9fb08709728d38553b93df6eed9417da1R23>.
Can you please update?

Take care,
Alicia
___________________________________
*Alicia M. Bentley, Ph.D.*
Global Verification Project Lead
Model Evaluation Group (MEG) <https://www.emc.ncep.noaa.gov/users/meg/home/>
NOAA/NWS/NCEP/EMC (VPPPG Branch)
Work Phone: ‪(301) 683-3783‬

**Comment by PerryShafran-NOAA** [comment]
Changed.  Thanks for noting that!  

Perry

**Comment by ShelleyMelchior-NOAA** [comment]
Thanks, @PerryShafran-NOAA ! I will begin testing this PR now. 
As in the analyses PR, can you change driver script lines like this one 
`source $HOMEevs/modulefiles/rtofs/rtofs_plots.sh`
 to
`source $HOMEevs/modulefiles/${COMPONENT}/${COMPONENT}_${STEP}.sh`

File: dev/drivers/scripts/rtofs/stats/jevs_rtofs_argo_stats.sh [hunk 1]
```diff
@@ -43,6 +29,8 @@ export VERIF_CASE=grid2obs
 export VARS="temp psal"
 export COMPONENT=rtofs
 
+source $HOMEevs/modulefiles/${COMPONENT}/${COMPONENT)_${STEP}.sh
```
Comment (ShelleyMelchior-NOAA) [inline dev/drivers/scripts/rtofs/stats/jevs_rtofs_argo_stats.sh]: One small typo that needs fixing ... the closing brace for the second COMPONENT needs to be curly.

**Comment by ShelleyMelchior-NOAA** [comment]
I have completed the run of rtofs prep. Please review.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/prep/rtofs` 
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_rtofs_prep.o82547193`

-----
But stats doesn't complete correctly b/c it doesn't automatically create $COMOUT. I have do my own mkdir of $COMOUT. We shouldn't require this. Can you please add `mkdir -p $COMOUT` to all of the JEVS_RTOFS_* j-jobs?

**Comment by ShelleyMelchior-NOAA** [comment]
Actually, I was a bit premature in my comment.
JEFS_RTOFS_PREP needs `mkdir -p $COMOUT $COMOUTprep`
JEVS_RTOFS_STATS needs `mkdir -p $COMOUT $COMOUTsmall $COMOUTfinal`
JEVS_RTOFS_*PLOTS is OK, I think, but could add $COMOUT to the existing `mkdir -p` command to be safe.

**Comment by PerryShafran-NOAA** [comment]
OK, all the fixes above should now be addressed.

I also added in the new ush file that will be needed for MET 11.1 and METplus 5.1.  We'll be testing that to see if it works.  We'll know right away after you run prep.  

Perry

**Comment by ShelleyMelchior-NOAA** [comment]
Ok. I have pulled in all of the changes and will re-run rtofs prep.

**Comment by PerryShafran-NOAA** [comment]
Further fixes in the modulefile and the closing bracket in all the stats driver scripts have been updated.

Perry

File: dev/drivers/scripts/rtofs/stats/jevs_rtofs_argo_stats.sh (hunk 1 above)
Comment (PerryShafran-NOAA) [reply dev/drivers/scripts/rtofs/stats/jevs_rtofs_argo_stats.sh]: Done.

**Comment by ShelleyMelchior-NOAA** [comment]
I cp'd the emc.vpppg prep dirs, back to June 20, to my $USER space. Then I ran rtofs prep keeping all COMIN/COMOUT variables directed to $USER. The prep processing ran to completion. Please review.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/prep/rtofs`
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_rtofs_prep.o82668634`

Next I will run stats using my local $USER prep data.

**Comment by ShelleyMelchior-NOAA** [comment]
@PerryShafran-NOAA , I tried to run aviso stats, but I am still having trouble. :-(
HOMEevs=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/githubwkspc/forks/PS/EVS`
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/rtofs`
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_rtofs_aviso_stats.o82750540`

**Comment by PerryShafran-NOAA** [comment]
Hi, @ShelleyMelchior-NOAA ,

You are running the prep for VDATE=20230711, but the stats for VDATE=20230712.  Make sure that these run for the same day - that would help get these to run properly.

Perry

**Comment by PerryShafran-NOAA** [comment]
Updates to move the processing to ptmp, as well as to create the masks in prep rather than stats, have all been added to the test branch.  You may re-test the prep and stats.   You should be sure to bring in the prep files from /lfs/h2/emc/vpppg/noscrub/perry.shafran/evs/v1.0/prep/rtofs to your directory before you run.

Perry

**Comment by ShelleyMelchior-NOAA** [comment]
@PerryShafran-NOAA , I re-ran prep last night, first copying `/lfs/h2/emc/vpppg/noscrub/perry.shafran/evs/v1.0/prep/rtofs` to `/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/prep/rtofs`. The ran completed over night, since it takes many hours to complete.
Output log file: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_rtofs_prep.o72596742`
If this looks OK, I will start stats and point to my own prep space. I will make sure the VDATEs match!

**Comment by PerryShafran-NOAA** [comment]
Everything looks great!  Now, be sure to check the stats runs to see that you are running for VDATE=20230718 to be sure that all looks OK.  Note that most of the runs will be quick, but argo and ndbc take longer (1 hr and 3 hr respectively).  I'll check it out once all runs complete.

Perry

**Comment by ShelleyMelchior-NOAA** [comment]
I ran stats for aviso, forcing VDATE to match that of prep. All ran to completion! Looks good! What do you think, Perry? I can start going through the rest of the stats next.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/rtofs/`
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_rtofs_aviso_stats.o72645040`

**Comment by ShelleyMelchior-NOAA** [comment]
I completed stats processing for all data sources: argo, aviso, ghrsst, ndbc, osisaf, smap, smos.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/rtofs`
logfiles:
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_rtofs_argo_stats.o72647127`
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_rtofs_aviso_stats.o72645040`
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_rtofs_ghrsst_stats.o72647247`
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_rtofs_ndbc_stats.o72647143`
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_rtofs_osisaf_stats.o72647251`
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_rtofs_smap_stats.o72647261`
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_rtofs_smos_stats.o72647266`

Things look good to  me. Please review.
I will hold off on plots testing until the fhr 3-digit situation is addressed or advised otherwise.

**Comment by PerryShafran-NOAA** [comment]
The stat files are all identical to mine and there seem to be no errors in any of the .o files.  

Thus, we are validated all the way through prep and stats.

Perry

**Comment by ShelleyMelchior-NOAA** [comment]
The rtofs plots runs have completed. Please review. I untarred one file (aviso grid2grid) and I see 3 digit forecast fields in the png file name.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/plots/rtofs`
log files:
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_rtofs_headline_plots.o72888041`
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_rtofs_plots.o72887983`

**Comment by ShelleyMelchior-NOAA** [comment]
I will complete the code review and approve the merge of this PR.

**Review by ShelleyMelchior-NOAA — APPROVED** [review]
This PR is approved for  merge.

**Pull request merged**